### PR TITLE
Create rake task for adding new Decidim Authority

### PR DESCRIPTION
**Pull request by gchacart**
This PR includes:
- Creation of the BB new app with postgresql.
- Setup of: Rubocop, Postgres, RSpec.
- Creation of a rake task for adding a new Decidim authority.
- Documentation for how to setup the new app.
- Documentation for how to add the new Decidimm authority.
- Tests for the new rake task.

📌 Related Issues
- Fixes https://github.com/decidim/decidim/issues/6288

📸 Screenshots

**Readme documentation:**
readme md

**Add authority documentation (located in docs/client/add_authority.md):**
add authority md

**Adding a new Decidim authority from console.** From a system console, being in the project folder, the execution command is:
console add authority command

### CODE REVIEW

File: db/migrate/20200717065508_create_clients.rb
Comment (slickepinne) [inline db/migrate/20200717065508_create_clients.rb:4]: I'm not 100% sure but maybe it makes sense to set `null: false` for the `name`, `public_key` and `api_key` as a client without any of these fields might not be valid. Also, regarding `unique: true`, I would add an index to these fields. 😊

File: lib/tasks/client.rake
Comment (agustibr) [inline lib/tasks/client.rake:22]: Could this be replaced with a [`find_or_initialize_by`](https://apidock.com/rails/v6.0.0/ActiveRecord/Relation/find_or_initialize_by)?

```rb
auth = Authority.find_or_initialize_by(name: name, public_key: public_key)

if auth.new_record?
  auth.update!(api_key: generate_api_key)
  puts "Authority '#{auth.name}' successfuly added!"
else
  puts "The authority already exists!"
end

puts "The API key for this authority is: #{auth.api_key.to_s}"
```

File: bin/bundle
Comment (leio10) [inline bin/bundle:43]: I'm not able to realize why is this line different from `gemfile.present?`. 🤔 

On the other hand, here you can use an inline comment to disable the rubocop rule only there, like this:
```ruby
return gemfile if gemfile && !gemfile.empty? # rubocop:disable Rails/Present
```

**Comment by leio10** [comment]
@tramuntanal @gchacart regarding the `.rubocop-https---raw-githubusercontent-com-decidim-...` files, they are automatically generated by rubocop after downloaded from github, so I think we should add them to the .gitignore file.

**Comment by tramuntanal** [comment]
@gchacart let us know when you've updated the .gitignore file and checked the review

File: README.md
Comment (gchacart) [reply README.md:28]: Hi! In Decidim we have this one:
https://github.com/decidim/decidim/blob/develop/LICENSE-AGPLv3.txt

Would it be exactly like that one?

Thank you.

File: docs/client/add_authority.md
Comment (gchacart) [reply docs/client/add_authority.md:6]: Yes, is a good idea! I'll put the types must be string. By now we are not sure about the length of the public key, so I won't add that info right now.

File: spec/tasks/client_spec.rb
Comment (gchacart) [reply spec/tasks/client_spec.rb:31]: I'm not sure about this one! It gives me an error.

File: bin/bundle
Comment (gchacart) [reply bin/bundle]: At some point, when executing rubocop it gave me an error about putting gemfile.present. Now I tried and works fine 👍

File: README.md
Comment (tramuntanal) [reply README.md]: Yes!
cc/ @decidim/product

File: docs/client/add_authority.md
Comment (tramuntanal) [reply docs/client/add_authority.md:10]: Regarding this, we must take into account the discussion here: https://github.com/decidim/decidim/pull/6420

**Comment by tramuntanal** [comment]
@gchacart is it ready?
@decidim/product should we wait to have the public key discussion in #6420 agreed before merging or we can merge and then rework this repo if it needs changes?
Also please check the license comment in my review :smile:

File: docs/client/add_authority.md
Comment (gchacart) [reply docs/client/add_authority.md:10]: Ok, I'll check that out

File: lib/tasks/client.rake
Comment (gchacart) [reply lib/tasks/client.rake]: I've done kind of a mixed solution here because I needed to ask with an OR and that's not possible with the `find_or_initialize_by` (at least I didn't find how)

**Comment by gchacart** [comment]
> @gchacart is it ready?
> @decidim/product should we wait to have the public key discussion in #6420 agreed before merging or we can merge and then rework this repo if it needs changes?
> Also please check the license comment in my review 😄

@tramuntanal Hi! Yes, I think we can merge and then review it if needed. Let me add the license and I'll let you know that I've finished. Thank you!

File: docs/client/add_authority.md
Comment (gchacart) [reply docs/client/add_authority.md:10]: As talked, we can take it into account for later

**Comment by andreslucena** [comment]
> I think we can merge and then review it if needed.

We on @decidim/product agree with this hypothesis, let's go ahead :rocket:

**Comment by gchacart** [comment]
> > I think we can merge and then review it if needed.
> 
> We on @decidim/product agree with this hypothesis, let's go ahead 🚀

@andreslucena Perfect, thank you!
@tramuntanal I already uploaded the License and we have product approval 😃

**Review by tramuntanal — APPROVED** [review]

**Pull request merged**
